src/predictor/gpu_predictor.cu

```diff
@@ -361,10 +371,9 @@ class GPUPredictor : public xgboost::Predictor {
       DeviceOffsets(batch.offset, batch.data.Size(), &device_offsets);
       batch.data.Reshard(GPUDistribution::Explicit(devices_, device_offsets));
 
-      // TODO(rongou): only copy the model once for all the batches.
       dh::ExecuteIndexShards(&shards_, [&](int idx, DeviceShard& shard) {
-        shard.PredictInternal(batch, dmat->Info(), out_preds, model,
-                              h_tree_segments, h_nodes, tree_begin, tree_end);
+        shard.PredictInternal(batch, dmat->Info(), out_preds, tree_begin, tree_end,
+                              model.param.num_output_group);
```

Comment:
I think all three of these parameters can be stored in the shard after `InitModel()`.

However, I'll leave this up to you.

Reply:
Done.